docs/plugins/verifier/cosign.md

```diff
@@ -1,23 +1,208 @@
+- A scope, which does NOT have a suffixed `*` character, is assumed to be an absolute reference. The entire scope must be an exact match to the image reference.
+
+There are restrictions on `scopes` listed within and across trust policies:
+
+- Scopes within or across trust policies CANNOT overlap in any form, regardless of it being an absolute scope or a scope using a wildcard character.
+  - INVALID: given scope-A = `myregistry.io/*` & scope-B = `myregistry.io/namespace/*`, these scopes overlap since scope-B is contained in scope-A
+  - INVALID: given scope-A = `myregistry.io/*` & scope-B = `myregistry.io/namespace/image@sha256:abcd1234`, these scopes overlap since absolute scope-B is contained in scope-A
+  - VALID: given scope-A = `myregistry.io/namespace1/*` & scope-B = `myregistry.io/namespace2/*`, these scopes do NOT overlap since neither scope can overlap with each other
+  - Why do we enforce strict scope overlap checks? To avoid unintended verification behaviors at verification time when scope matching occurs. To set a higher security bar, Ratify makes sure that every single scope is guaranteed to match to a single trust policy. This validation occurs on creation of the verifier and NOT at verification time, thus allowing users to catch misconfigurations ahead of time.
+
+The single wildcard `*` scope is a special global scope that encompasses ALL references. Only a single trust policy can define a `*` global scope. If `*` scope exists in conjunction with other trust policies containing more granular scopes, the trust policy with a more granular matching scope will be selected instead.
+
+>WARNING: Ratify, by default, enables tag-to-digest mutation. All tag based scopes will NOT match to an intended scope. Either scope to a specific digest or provide a wild card scope such as `myregistry.io/namespace/image*`
+
+### Keys
+
+A trust policy can be configured with a list of `keys` that are trusted for a particular policy. Each entry in the list of `keys` corresponds to either: all the keys in a particular `KeyManagementProvider` resource OR a specific key when a `name` and optionally a `version` is provided.
+
+The `provider` field is always required except for when the `file` field is defined. The `provider` is the name of the `KeyManagementProvider` resource Ratify should look for configured keys from. If the `name` field is not provided for a specific key, all keys in the `KeyManagementProvider` are trusted.
+
+The `name` field specifies a specific key defined in the provider. An optional `version` can be defined if the latest version is not desired.
+
+The `file` field defines an absolute file path to a local public key. This field should NOT be used in conjunction with `provider`, `name`, `version`.
+
+#### Supported Key Types
+
+- RSA: 2048, 3072, 4096
+- EC: P256, P384, P521
+- ED25519
+
+### Limitations
+
+Currently, Cosign trust policies only support key-based configurations. Keyless support will be added soon.
```

Comment:
should we create an issue for tracking?

Reply:
Yes I've been using https://github.com/deislabs/ratify/issues/1323